### PR TITLE
feat(sponsorblock): support actionTypes

### PR DESCRIPTION
**Pull request by FineFindus**
Adds support for requesting different `actionType`s when requesting SponsorBlock segments.

### CODE REVIEW

File: src/main/java/me/kavin/piped/utils/SponsorBlockUtils.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public static String getSponsors(String id, String categories)
             try {
 
                 var resp = RequestUtils.sendGetRaw(url + "/api/skipSegments/" + URLUtils.silentEncode(hash.substring(0, 4))
-                        + "?categories=" + URLUtils.silentEncode(categories)).get();
+                        + "?categories=" + URLUtils.silentEncode(categories) + "&actionTypes=" + URLUtils.silentEncode(actionType)).get();
```
Comment (FireMasterK) [inline src/main/java/me/kavin/piped/utils/SponsorBlockUtils.java:37]: We should also handle when the `actionType` is null.

File: src/main/java/me/kavin/piped/utils/SponsorBlockUtils.java [hunk 2]
```diff
@@ -22,19 +22,22 @@
...
 
         if (StringUtils.isEmpty(categories))
             return mapper.writeValueAsString(new InvalidRequestResponse());
 
         String hash = DigestUtils.sha256Hex(id);
 
-        for (String url : Constants.SPONSORBLOCK_SERVERS) {
+        for (String apiUrl : Constants.SPONSORBLOCK_SERVERS) {
             try {
+                String url =apiUrl + "/api/skipSegments/" + URLUtils.silentEncode(hash.substring(0, 4))
```
Comment (Bnyro) [inline src/main/java/me/kavin/piped/utils/SponsorBlockUtils.java:35]:
```suggestion
                String url = apiUrl + "/api/skipSegments/" + URLUtils.silentEncode(hash.substring(0, 4))

```
nit

**Review by Bnyro — APPROVED** [review]
Thanks :+1:

**Comment by FineFindus** [comment]
Thank you. Uh, now I just have to remember why I implemented this... :)

**Pull request merged**